ui/src/components/btn/__tests__/use-btn.spec.js

```diff
@@ -1,187 +1,513 @@
+import { mount } from '@cypress/vue'
+import { createMemoryHistory, createRouter } from 'vue-router'
+import { alignMap, alignValues } from '../../../composables/private/use-align.js'
+import QBtn from '../QBtn.js'
+import { btnPadding as paddingMap } from '../use-btn.js'
+
+const defaultOptions = {
+  label: 'simple Btn'
+}
+
+const typesValues = [ 'button', 'submit', 'reset' ]
+
+const paddingValues = Object.keys(paddingMap)
+
+function mountQBtn (options = {}) {
+  // Setup options object
+  options.global = options.global || {}
+  options.global.plugins = options.global.plugins || []
+
+  options.props = {
+    ...defaultOptions,
+    ...options.props
+  }
+
+  if (!options.router) {
+    options.router = createRouter({
+      routes: [],
+      history: createMemoryHistory()
+    })
+  }
+
+  // Add router plugin
+  options.global.plugins.push({
+    install (app) {
+      app.use(options.router)
+    }
+  })
+
+  return mount(QBtn, options)
+}
+
 describe('use-btn API', () => {
   describe('Props', () => {
     describe('Category: behavior|state', () => {
       describe('(prop): loading', () => {
-        it.skip(' ', () => {
-          //
+        it('should render a button with "loading" slot when "loading" prop is set to true', () => {
+          mountQBtn({
+            props: {
+              loading: true
+            }
+          })
+
+          cy.get('.q-btn')
+            .get('.absolute-full.flex.flex-center').should('exist')
         })
       })
     })
 
     describe('Category: content', () => {
       describe('(prop): label', () => {
-        it.skip(' ', () => {
-          //
+        it('should render a label inside the button', () => {
+          const label = 'Custom Label'
+          mountQBtn({
+            props: {
+              label
+            }
+          })
+
+          cy.get('.q-btn').should('contain', label)
         })
       })
 
       describe('(prop): icon', () => {
-        it.skip(' ', () => {
-          //
+        it('should render an icon on the left', () => {
+          const icon = 'home'
+
+          mountQBtn({
+            props: {
+              icon
+            }
+          })
+
+          cy.get('.q-btn').should('contain', icon)
         })
       })
 
       describe('(prop): icon-right', () => {
-        it.skip(' ', () => {
-          //
+        it('should render an icon on the right', () => {
+          const iconRight = 'home'
+
+          mountQBtn({
+            props: {
+              iconRight
+            }
+          })
+
+          cy.get('.q-btn').should('contain', iconRight)
```

Comment:
It would need a more specific class which makes sure the icon is actually on the right